siac/rentercmd.go

```diff
@@ -266,24 +266,45 @@ func renterallowancecancelcmd() {
 }
 
 // rentersetallowancecmd allows the user to set the allowance.
-func rentersetallowancecmd(amount, period, hosts, renewWindow string) {
-	hastings, err := parseCurrency(amount)
+// the first two parameters, amount and period, are required.
+// the second two parameters are optional:
+//    hosts                 integer number of hosts
+//    renewperiod           how many blocks between renewals
+func rentersetallowancecmd(cmd *cobra.Command, args []string) {
+	if len(args) == 0 {
+		die("Must supply amount (first) argument")
+	}
+	hastings, err := parseCurrency(args[0])
 	if err != nil {
 		die("Could not parse amount:", err)
 	}
-	blocks, err := parsePeriod(period)
+	if len(args) == 1 {
+		die("Must supply period (second) argument")
+	}
+	blocks, err := parsePeriod(args[1])
 	if err != nil {
 		die("Could not parse period")
 	}
-	_, err = strconv.Atoi(hosts)
-	if err != nil {
-		die("Count not parse host count")
+	queryString := fmt.Sprintf("funds=%s&period=%s", hastings, blocks)
+	if len(args) > 2 {
+		_, err = strconv.Atoi(args[2])
+		if err != nil {
+			die("Could not parse host count")
+		}
+		queryString += fmt.Sprintf("&hosts=%s", args[2])
 	}
-	_, err = parsePeriod(renewWindow)
-	if err != nil {
-		die("Could not parse renew window")
+	if len(args) > 3 {
+		renewWindow, err := parsePeriod(args[3])
+		if err != nil {
+			die("Could not parse renew window")
+		}
+		queryString += fmt.Sprintf("&renewwindow=%s", renewWindow)
+	}
+	if len(args) > 4 {
+		cmd.UsageFunc()(cmd)
+		os.Exit(exitCodeUsage)
```

Comment:
@lukechampine is this the preferred response when the wrong number of commands are used?